library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -289,34 +289,62 @@ class Client() {
         }
     }
 
-    // This is a V3 only feature
-    suspend fun createOrBuild(
+    private suspend fun initializeV3Client(
+        accountAddress: String,
+        clientOptions: ClientOptions,
+        signingKey: SigningKey? = null,
+    ): Client {
+        val inboxId = getOrCreateInboxId(clientOptions, accountAddress)
+
+        val (libXMTPClient, dbPath) = ffiXmtpClient(
+            clientOptions,
+            signingKey,
+            clientOptions.appContext,
+            null,
+            accountAddress,
+            inboxId
+        )
+
+        libXMTPClient?.let { client ->
+            return Client(
+                accountAddress,
+                client,
+                dbPath,
+                client.installationId().toHex(),
+                client.inboxId(),
+                clientOptions.api.env
+            )
+        } ?: throw XMTPException("Error creating V3 client: libXMTPClient is null")
+    }
+
+    // Function to create a V3 client with a signing key
+    suspend fun createV3(
         account: SigningKey,
-        options: ClientOptions,
+        options: ClientOptions? = null,
     ): Client {
         this.hasV2Client = false
-        val inboxId = getOrCreateInboxId(options, account.address)
-
+        val clientOptions = options ?: ClientOptions(enableV3 = true)
+        val accountAddress =
+            if (account.isSmartContractWallet) "eip155:${account.chainId}:${account.address.lowercase()}" else account.address.lowercase()
         return try {
-            val (libXMTPClient, dbPath) = ffiXmtpClient(
-                options,
-                account,
-                options.appContext,
-                null,
-                account.address,
-                inboxId
-            )
+            initializeV3Client(accountAddress, clientOptions, account)
+        } catch (e: Exception) {
+            throw XMTPException("Error creating V3 client: ${e.message}", e)
+        }
+    }
 
-            libXMTPClient?.let { client ->
-                Client(
-                    account.address,
-                    client,
-                    dbPath,
-                    client.installationId().toHex(),
-                    client.inboxId(),
-                    options.api.env
-                )
-            } ?: throw XMTPException("Error creating V3 client: libXMTPClient is null")
+    // Function to build a V3 client without a signing key (using only address (& chainId for SCW))
+    suspend fun buildV3(
+        address: String,
+        contractChainId: Long? = null,
```

Comment:
Do people know what scw stand for? I would be okay with calling this `scwChainId` if we don't think it's confusing.